tests/rptest/tests/acls_test.py

```diff
@@ -321,6 +321,55 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            # run a few times for good health
+            for _ in range(2):
+                try:
+                    self.get_client("base").acl_list()
+                    raise RuntimeError(
+                        'list acls should have failed with base user')
+                except ClusterAuthorizationError:
+                    pass
+
+                # should succeed
+                self.get_client("cluster_describe").acl_list()
+                self.get_super_client().acl_list()
+
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        check_perms()
+        '''
+        Change the authn method and initiate feature migration
+        with a rolling restart. Do not recreate tls certs.
+        '''
+        self.security.endpoint_authn_method = authn_method
+        self.redpanda.set_security_settings(self.security, init_tls=False)
+        self.logger.debug('Initiating rolling restart')
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            start_timeout=90,
+                                            stop_timeout=90)
+        '''
+        Once restart is complete, change cluster wide configs for
+        kafka_enable_authorization and kafka_mtls_principal_mapping_rules
+        via the admin api. Do not expect a restart.
+
+        NOTE:
+        - kafka_enable_authorization overrides sasl in an authorization context
+        - authentication_method overrides sasl in an authentication context
+        '''
+        new_cfg = {'kafka_enable_authorization': True}
+        if authn_method == 'mtls_identity':
+            new_cfg['kafka_mtls_principal_mapping_rules'] = [
+                'RULE:.*CN=([^,]+).*/$1/', 'DEFAULT'
+            ]
+        self.redpanda.set_cluster_config(new_cfg)
+
+        check_perms()
```

Comment:
to make this test stronger, it seems like we could add a check in between rolling restart and set cluster config. in short, how does the test ensure that check_perms is operating based on new settings and not on whatever allowed check_perms to succeed prior to the restart?

Reply:
Fair point. I thought of a similar argument overnight. I'll make the change.

Reply:
Done.